File: iCareApp/app/src/main/java/com/lanthanh/admin/icareapp/presentation/base/BaseFragment.java
package com.lanthanh.admin.icareapp.presentation.base;

import android.support.v4.app.Fragment;
import android.widget.Toast;

/**
 * @author longv
 * Created on 19-Mar-17.
 */

public abstract class BaseFragment<T extends BasePresenter> extends Fragment{
    private Toast toast;
    public abstract void initViews();
    public abstract void refreshViews();
    public  T getMainPresenter() {
        T t = null;
        return t;
    }
    public void showMessage(String msg){
        Toast toast = Toast.makeText(getActivity(), msg, Toast.LENGTH_SHORT);
        toast.show();
    }
}

File: iCareApp/app/src/main/java/com/lanthanh/admin/icareapp/presentation/base/BaseFragment.kt
package com.lanthanh.admin.icareapp.presentation.base

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.View
import android.widget.Toast

/**
 * @author longv
 * * Created on 19-Mar-17.
 */

abstract class BaseFragment<T : Presenter> : Fragment() {
    abstract fun initViews()
    abstract fun refreshViews()
    abstract fun getMainPresenter() : T

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initViews()
    }

    fun showToast(msg: String) {
        val toast = Toast.makeText(activity, msg, Toast.LENGTH_SHORT)
        toast.show()
    }
}
